### ISSUE TITLE
strict: false option on Model#update no effect..?

### ISSUE THREAD

**Issue by MidnightP** [Automattic/mongoose#5111]
My models do not have the strict option on them so by default strict is true. I want to store fields, some of which are defined in the schema, and some of which are not. So I presume to use `strict: false`, like so:

```
const options = { 
                upsert: true,
		runValidators: false,
                strict: false }

const data = { field_one: 'blablabla', field_two: 'blablalblalblalbl' }

Model.update({_id: _id}, data, options, cb)
```

Now, `field_one` is in the schema, `field_two` is not. When I look at the document using a `.find` or using MongoChef, i see that the field is not there!

Temporarily brutal solution:

1) `Model.schema.options.strict = false`
2) perform update
3) `Model.schema.options.strict = true` 

**Do you want to request a *feature* or report a *bug*?**

I think so.

**What is the current behavior?**

`field_one` is saved & `field_two` is not.

**What is the expected behavior?**

Overriding the strict option of the schema!

**Please mention your node.js, mongoose and MongoDB version.**

Mongoose @4.9.0
Node @7.6.0
Mongodb@2.2.4

I'm running mongodb in Docker with image `mongo:3.4`

**Comment by vkarpov15** [comment]
So I found a related issue where `fieldTwo` would not show up in the mongoose doc that should be fixed with f4d7ad6, but the below script works for me (bypassing mongoose's findOne() and using the underlying mongodb driver's findOne() ). The update getting sent down looks correct. Can you modify the below script to reproduce your issue?

```javascript
var assert = require('assert');
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

mongoose.set('debug', true);
mongoose.connect('mongodb://localhost:27017/gh5111');

var schema = new Schema({
  fieldOne: String
}, { strict: true });
var Test = mongoose.model('gh5111', schema);

Test.create({ fieldOne: 'Test' }).
  then(function() {
    var data = { fieldOne: 'Test2', fieldTwo: 'Test3' };
    var opts = {
      upsert: true,
      runValidators: false,
      strict: false
    };
    return Test.update({}, data, opts);
  }).
  then(function() {
    return Test.collection.findOne();
  }).
  then(function(doc) {
    assert.equal(doc.fieldOne, 'Test2');
    assert.equal(doc.fieldTwo, 'Test3');
    process.exit(0);
  }).
  catch(function(error) {
    console.error(error);
    process.exit(-1);
  });
```

**Comment by MidnightP** [comment]
Okay, I can't reproduce it here.

If I ever see it again i will report. Thanks for the effort!

Something I did find though is that I seem to need to `.lean()` the `.find()` in order to get out all the fields...

Isn't that weird?

Please see the note below:

`var assert = require('assert')
var mongoose = require('mongoose')
var Schema = mongoose.Schema

mongoose.set('debug', true);
mongoose.connect('mongodb://localhost:27017/gh5111')

var schema = new Schema({
  fieldOne: String
}, { strict: true })
var Test = mongoose.model('gh5111', schema);

  let _id

  Test.collection.drop()

  var data = { fieldOne: 'Test2' }

  Test.create(data, opts).
  then(function() {
    return Test.findOne()
  }).
  then(function(doc) {
    _id = doc._id
    assert.equal(doc.fieldOne, 'Test2')
  }).
  then(function() {
    var data = { fieldTwo: 'Test3' }
    var opts = {
      upsert: true,
      runValidators: false,
      strict: false
    }
    return Test.update({ _id }, data, opts)
  }).
  then(function() {
    // NOTE If I don't use findOne().lean()
    // then below fieldTwo is in the document (verify by logging doc it) but doc.fieldTwo returns undefined....
    return Test.findOne()
  }).
  then(function(doc) {
    console.log(doc)
    assert.equal(doc.fieldOne, 'Test2')
    assert.notEqual(doc.fieldTwo, 'Test3')
  }).
  then(function() {
    return Test.findOne().lean()
  }).
  then(function(doc) {
    assert.equal(doc.fieldOne, 'Test2')
    assert.equal(doc.fieldTwo, 'Test3')
    process.exit(0)
  }).
  catch(function(error) {
    console.error(error)
    process.exit(-1)
  })
`
